pkg/minishift/addon/parser/addon_parser.go

```diff
@@ -70,42 +70,47 @@ func NewAddOnParser() *AddOnParser {
 	return &parser
 }
 
-// Parse takes as parameter a reader containing an addon definition and returns an AddOn instance.
-// If an error occurs, the error is returned.
-func (parser *AddOnParser) Parse(addOnDir string) (addon.AddOn, error) {
-	addonInstallReader, err := parser.getAddOnContentReader(addOnDir, ".addon")
-	if err != nil {
-		return nil, err
-	}
-	addonRemoveReader, err := parser.getAddOnContentReader(addOnDir, ".addon.remove")
+func (parser *AddOnParser) getAddOnContent(addOnDir, fileSuffix string) (addon.AddOnMeta, []command.Command, error) {
+	var (
+		meta     addon.AddOnMeta
+		commands []command.Command
+		err      error
+	)
+
+	addonReader, err := parser.getAddOnContentReader(addOnDir, fileSuffix)
 	if err != nil {
-		return nil, err
+		return nil, nil, err
 	}
-	meta, commands, err := parser.parseAddOnContent(addonInstallReader)
-	if err != nil {
+	if addonReader != nil {
+		meta, commands, err = parser.parseAddOnContent(addonReader)
 		name := ""
 		if meta != nil {
 			name = meta.Name()
 		}
-		return nil, NewParseError(err.Error(), name, addOnDir)
-	}
-
-	var removeCommands []command.Command
-	var metaDataForAddonRemove addon.AddOnMeta
-	if addonRemoveReader != nil {
-		metaDataForAddonRemove, removeCommands, err = parser.parseAddOnContent(addonRemoveReader)
 		if err != nil {
-			name := ""
-			if metaDataForAddonRemove != nil {
-				name = metaDataForAddonRemove.Name()
-			}
-			return nil, NewParseError(err.Error(), name, addOnDir)
+			return nil, nil, NewParseError(err.Error(), name, addOnDir)
+		}
+		if filepath.Base(addOnDir) != name {
+			return nil, nil, NewParseError(fmt.Sprintf("Add-on directory name should match to addon name"), addOnDir, "")
```

Comment:
This should be `return nil, nil, NewParseError("Add-on directory name should match to addon name","", addOnDir)` because you are not doing any formatting and `NewParseError` definition says `NewParseError(msg string, name string, dir string)` so `addOnDir` sould be last one.